internal/mq/mqimpl/natsmq/server/global_nmq.go

```diff
@@ -0,0 +1,88 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/nats-io/nats-server/v2/server"
+	"go.uber.org/zap"
+
+	"github.com/milvus-io/milvus/pkg/log"
+	"github.com/milvus-io/milvus/pkg/util/paramtable"
+)
+
+// Nmq is global natsmq instance that will be initialized only once
+var Nmq *server.Server
+
+// once is used to init global natsmq
+var once sync.Once
+
+// InitNatsMQ init global natsmq single instance
+func InitNatsMQ(storeDir string) error {
+	var finalErr error
+	once.Do(func() {
+		opts, initializeTimeout := getServerOption(storeDir)
+		log.Debug("try to initialize global nmq", zap.Reflect("options", opts), zap.Duration("timeout", initializeTimeout), zap.String("storeDir", storeDir))
+		Nmq, finalErr = server.NewServer(opts)
+		log.Debug("initialize nmq finished", zap.Error(finalErr))
+		if finalErr != nil {
+			return
+		}
+		go Nmq.Start()
+		// Wait for server to be ready for connections
+		if !Nmq.ReadyForConnections(initializeTimeout) {
+			finalErr = fmt.Errorf("nmq is not ready with timeout %s", initializeTimeout)
+			log.Warn("nmq is not ready", zap.Duration("timeout", initializeTimeout), zap.Reflect("options", opts))
```

Comment:
It might cause some exception behaviors if nmq  server is not ready and skip it within a singleton.

Reply:
new commit changed:
log.Fatal with error field is added into `MustInitNatsMQ`